posthog/temporal/batch_exports/spmc.py

```diff
@@ -628,30 +656,51 @@ async def produce_batch_export_record_batches_from_range(
                 should be skipped.
             queue: The queue where to produce record batches.
             query_parameters: Additional query parameters.
+            team_id: The team ID of the batch export.
+            is_backfill: Whether the batch export is a backfill.
             max_record_batch_size_bytes: The max size in bytes of a record batch to insert in `queue`.
                 If a record batch is larger than this, `slice_record_batch` will be used to slice it
                 into smaller record batches.
             min_records_batch_per_batch: If slicing a record batch, each slice should contain at least
                 this number of records.
         """
-        for interval_start, interval_end in generate_query_ranges(full_range, done_ranges):
-            if interval_start is not None:
-                query_parameters["interval_start"] = interval_start.strftime("%Y-%m-%d %H:%M:%S.%f")
-            query_parameters["interval_end"] = interval_end.strftime("%Y-%m-%d %H:%M:%S.%f")
-            query_id = uuid.uuid4()
-
-            try:
-                async for record_batch in self.clickhouse_client.astream_query_as_arrow(
-                    query, query_parameters=query_parameters, query_id=str(query_id)
-                ):
-                    for record_batch_slice in slice_record_batch(
-                        record_batch, max_record_batch_size_bytes, min_records_per_batch
+        clickhouse_url = None
+        # 5 min batch exports should query a single node, which is known to have zero replication lag
+        if is_5_min_batch_export(full_range=full_range):
+            clickhouse_url = settings.CLICKHOUSE_OFFLINE_5MIN_CLUSTER_HOST
+
+        # Data can sometimes take a while to settle, so for 5 min batch exports we wait several seconds just to be safe.
+        # For all other batch exports we wait for 1 minute since we're querying the events_recent table using a
+        # distributed table and setting `max_replica_delay_for_distributed_queries` to 1 minute
+        if is_5_min_batch_export(full_range):
+            delta = dt.timedelta(seconds=30)
+        else:
+            delta = dt.timedelta(minutes=1)
+        end_at = full_range[1]
+        await wait_for_delta_past_data_interval_end(end_at, delta)
+
+        async with get_client(team_id=team_id, clickhouse_url=clickhouse_url) as client:
```

Comment:
I like this move, there is more configuration stuff we can shuffle around some more. Good work.